chainbase/src/main/java/org/tron/core/db2/common/TxCacheDB.java

```diff
@@ -42,6 +56,7 @@ public class TxCacheDB implements DB<byte[], byte[]>, Flusher {
   private BloomFilter<byte[]>[] bloomFilters = new BloomFilter[2];
   // filterStartBlock record the start block of the active filter
   private volatile long filterStartBlock = INVALID_BLOCK;
+  private volatile long currentBlockNum = INVALID_BLOCK;
```

Comment:
A check for  `currentBlockNum` with `block header` is needed? 

Node shutdown using `Kill -9` may result in DBS being inconsistent, so this PR only works with a graceful shutdown?  Just for a confirm

Reply:
> A check for currentBlockNum with block header is needed?

  Good question, but how might it be inconsistent?  Yes ,a graceful shutdown is required

Reply:
Two questions, with no relation. `A check` just for a double check, seems it introduces no more benefit.


> how might it be inconsistent

The whole shutdown process is complicated, still thinking

Reply:
Maybe the empty blocks will make different, a check is not required.